agent/config.go

```diff
@@ -2127,7 +2127,7 @@ func ReadConfigPaths(paths []string) (*Config, error) {
 // and performs go-sockaddr/template Parse on each known address in case the
 // user specified a template config for any of their values.
 func (c *Config) ResolveTmplAddrs() (err error) {
-	parse := func(addr *string, name string) {
+	parse := func(addr *string, validateIP bool, name string) {
```

Comment:
I think you want to validate the ip address if the value is not a unix socket path. What this parameter should express is whether a unix socket path is allowed for this value, e.g. `socketAllowed bool` instead of `validateIP`. With `socketAllowed` set to true you could also check whether the format of the UNIX socket path is correct and fail if it isn't.